components/resizable_sidebar/resizable_rhs/index.tsx

```diff
@@ -0,0 +1,215 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {HTMLAttributes, useCallback, useEffect, useLayoutEffect, useMemo, useRef, useState} from 'react';
+import {useSelector} from 'react-redux';
+
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/common';
+
+import {getRhsSize} from 'selectors/rhs';
+import LocalStorageStore from 'stores/local_storage_store';
+import {RHS_MIN_MAX_WIDTH, SidebarSize} from 'utils/constants';
+import {isOverLimit, isResizableSize, requestAnimationFrameForMouseMove, shouldRhsOverlapChannelView, shouldSnapWhenSizeGrown, shouldSnapWhenSizeShrunk} from '../utils';
+
+interface Props extends HTMLAttributes<'div'> {
+    children: React.ReactNode;
+}
+
+function Resizable({
+    role,
+    children,
+    id,
+    className,
+}: Props, ref: React.Ref<HTMLDivElement>) {
+    const forwardRef = ref as React.RefObject<HTMLDivElement>;
+    const rhsRef = useRef<HTMLDivElement>(null);
+    const resizeLineRef = useRef<HTMLDivElement>(null);
+
+    const [isResizeLineSelected, setIsResizeLineSelected] = useState(false);
+    const previousClientX = useRef(0);
+
+    const rhsSize = useSelector(getRhsSize);
+    const userId = useSelector(getCurrentUserId);
+
+    const minWidth = useMemo(() => RHS_MIN_MAX_WIDTH[rhsSize].min, [rhsSize]);
+    const maxWidth = useMemo(() => RHS_MIN_MAX_WIDTH[rhsSize].max, [rhsSize]);
+    const defaultWidth = useMemo(() => RHS_MIN_MAX_WIDTH[rhsSize].default, [rhsSize]);
+
+    const isRhsResizable = useMemo(() => isResizableSize(rhsSize), [rhsSize]);
+    const shouldRhsOverlap = useMemo(() => shouldRhsOverlapChannelView(rhsSize), [rhsSize]);
+
+    const handleDoubleClick = useCallback(() => {
+        if (rhsRef.current) {
+            rhsRef.current.style.width = `${defaultWidth}px`;
+            rhsRef.current.classList.add('sidebar--right-double-clicked');
+
+            setTimeout(() => {
```

Comment:
I think we might want to use `requestAnimationFrame` here instead of `setTimeout` as well as another place a few lines below this.

This feels like one of the few times where that name makes sense with how we're using it because we're actually trying to skip an animation here 😀

Reply:
I also think it's a perfect case to use :)